frame/scheduler/src/lib.rs

```diff
@@ -802,6 +802,9 @@ impl<T: Config> Pallet<T> {
 			if let Some(id) = s.maybe_id {
 				Lookup::<T>::remove(id);
 			}
+			if !Agenda::<T>::get(when).iter().any(|i| i.is_some()) {
```

Comment:
If you read the issue again; we can be even smarter here:  
> Scheduler should remove any trailing None items in agendas as items get removed.

Since your logic now would not truncate `[Some(_), None]` to `[Some(_)]`.

Reply:
update the impl and one test to check the trailing None case